File: on-boarding-samples/on-boarding-sample-java/src/main/java/apio/architect/example/CredentialsProvider.java
package apio.architect.example;


import com.liferay.apio.architect.credentials.Credentials;
import com.liferay.apio.architect.provider.Provider;
import org.osgi.service.component.annotations.Component;

import javax.servlet.http.HttpServletRequest;

@Component
public class CredentialsProvider implements Provider<Credentials> {

    @Override
    public Credentials createContext(HttpServletRequest httpServletRequest) {
        return () -> "";
    }

}

File: on-boarding-samples/on-boarding-sample-kotlin/src/main/java/apio/architect/example/CredentialsProvider.kt
package apio.architect.example

import com.liferay.apio.architect.credentials.Credentials
import com.liferay.apio.architect.provider.Provider
import org.osgi.service.component.annotations.Component

import javax.servlet.http.HttpServletRequest

@Component
class CredentialsProvider : Provider<Credentials> {

    override fun createContext(httpServletRequest: HttpServletRequest) = Credentials { "" }

}